### PR TITLE
Add expenditure layout

### PR DESCRIPTION
**Pull request by agnieszkajarosikloj**
This PR adds the routing, and layout for the new Expenditures page

### CODE REVIEW

File: src/modules/dashboard/components/ColonyHome/ColonyHomeLayout.tsx [hunk 1]
```diff
@@ -88,6 +101,17 @@ const ColonyHomeLayout = ({
                     ethDomainId={ethDomainId}
                   />
                 )}
+                {showExpenditure && (
+                  <Button
+                    appearance={{ theme: 'primary', size: 'large' }}
+                    text={MSG.newExpenditure}
+                    onClick={() =>
+                      history.push(
+                        `/colony/${colony.colonyName}/create-expenditure`,
```
Comment (jdakowicz) [inline src/modules/dashboard/components/ColonyHome/ColonyHomeLayout.tsx:110]: Can we use the `~routes/index'` and `generatePath()` from `react-router`?

```
// add this to the routes file
COLONY_EXPENDITURE_ROUTE = '/colony/:name/create/expenditure'

// use it like this
generatePath(COLONY_EXPENDITURE_ROUTE, {name: colony.colonyName})
```

File: src/modules/dashboard/components/ColonyHome/ColonyNavigation/ColonyNavigation.tsx [hunk 2]
```diff
@@ -75,6 +79,12 @@ const ColonyNavigation = ({ colony: { colonyAddress, colonyName } }: Props) => {
         text: MSG.linkTextExtensions,
         dataTest: 'extensionsNavigationButton',
       },
+      {
+        exact: false,
+        linkTo: `/colony/${colonyName}/expenditure`,
```
Comment (jdakowicz) [inline src/modules/dashboard/components/ColonyHome/ColonyNavigation/ColonyNavigation.tsx:84]: Check my previous comment about url

File: src/modules/pages/ExpenditurePage/ExpenditureTemplate.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import React, { ReactNode } from 'react';
+
+import { getMainClasses } from '~utils/css';
+
+import styles from './ExpenditureTemplate.css';
+
+interface Props {
```
Comment (jdakowicz) [inline src/modules/pages/ExpenditurePage/ExpenditureTemplate.tsx:7]: Do we need this? 
Can't we just use the `React.FC` type?

**Review by jdakowicz — CHANGES_REQUESTED** [review]

**Review by jdakowicz — APPROVED** [review]

File: src/modules/pages/components/ExpenditurePage/ExpenditureTemplate.css [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  width: 100%;
+  background-color: var(--colony-white);
+}
+
+.mainContent {
+  height: 100%;
+}
+
+.sidebar {
+  height: 100vh;
```
Comment (arrenv) [inline src/modules/pages/components/ExpenditurePage/ExpenditureTemplate.css:20]: @agnieszkajarosikloj Can we have the header subtracted from this, so we don't have the page scroll when not needed.

File: src/routes/routeConstants.ts [hunk 5]
```diff
@@ -21,3 +22,4 @@ export const ACTIONS_PAGE_ROUTE = `${COLONY_HOME_ROUTE}/tx/:transactionHash`;
 export const COIN_MACHINE_ROUTE = `${COLONY_HOME_ROUTE}/buy-tokens/:transactionHash?`;
 export const UNWRAP_TOKEN_ROUTE = `${COLONY_HOME_ROUTE}/unwrap-tokens`;
 export const CLAIM_TOKEN_ROUTE = `${COLONY_HOME_ROUTE}/claim-tokens`;
+export const EXPENDITURE_ROUTE = `${COLONY_HOME_ROUTE}/create-expenditure`;
```
Comment (arrenv) [inline src/routes/routeConstants.ts:25]: @agnieszkajarosikloj Can we please go with `/expenditure/create`, then when an expenditure has a unique URL it goes to `/expenditure/{UID}`

**Review by arrenv — CHANGES_REQUESTED** [review]
Great start @agnieszkajarosikloj, thank you.

Only a couple of comments.

File: src/styles/variables.css [hunk 6]
```diff
@@ -38,6 +38,8 @@
   --temp-grey-11: rgb(136, 136, 136); /* Used for the text color of time for the 'CardActivity' items */
   --temp-grey-12: rgb(152, 164, 193); /* Used for Drag & Drop placeholder */
   --temp-grey-13: rgb(227, 224, 227); /* Used for separator background */
+  --temp-grey-14: rgb(244, 241, 244); /* Used for top nav background */
+  --temp-grey-15: rgba(60, 68, 77, 0.2); /* Used for top nav border */
```
Comment (chinins) [inline src/styles/variables.css:42]: I am not sure that we should add the colours to the var file if they are just used once... I understand that it's a core element as a pro to do that but still not convinced.
I don't actually understand why we are doing a different border and background colours for this nav bar vs. what we already have for the actions/motions page. i see that you implemented the spec - this is a question to @arren.

**Review by chinins — COMMENTED** [review]
Nice work! Several notes:
- Please, have a look at our PRs descriptions and the branches naming conventions and follow them accordingly. The goal is to help reviewers to understand what to expect from a PR.
- each PR should correspond to an issue, and we add the issue number in the PR name, e.g.: `feature/4050-add-expenditure-layout`
- No need to add `feature`, `refactor` etc in the PR name, just something that's a bit more descriptive to humans
- Please, provide a descriptions (at least a short one) for the reviewers to understand what to expect, maybe a prit screen, a link to specific figma screen. At the bottom we always add `resolves#<issue number>' so it provides a link to the issue, where we can look for more details as well
- it would be good to have a PR for the  feature branch `feature/advanced-expenditures` where we can see all of the work that has already been done (obviously more for the future) and also some description and links for easy access

These are PR practices that will help us and hopefully you to be more efficient and to improve communication.

@arrenv, question to you about the sidebar colour (repeated here from one of my comments):
Why we are doing a different border and background colours for this nav bar vs. what we already have for the actions/motions page? Shouldn't we have something consistent across the dapp, especially for a core element like navbar?

**Comment by arrenv** [comment]
> @arrenv, question to you about the sidebar colour (repeated here from one of my comments): Why we are doing a different border and background colours for this nav bar vs. what we already have for the actions/motions page? Shouldn't we have something consistent across the dapp, especially for a core element like navbar?

Yes, great pickup. @agnieszkajarosikloj please use the exact same header background and border as what is on the actions page. If you click one of the items in the actions list, it will take you to the actions page so that you can see it.

File: src/styles/variables.css (hunk 6 above)
Comment (arrenv) [reply src/styles/variables.css:42]: Yes, good pickup. It is best to use what is already in the DApp. Thanks @chinins!

File: src/utils/hooks/useTitle.ts [hunk 7]
```diff
@@ -134,6 +141,10 @@ const MSG = defineMessages({
     id: 'utils.hooks.useTitle.fallbackTitle',
     defaultMessage: `Colony`,
   },
+  createExpenditure: {
+    id: 'utils.hooks.useTitle.createExpenditure',
+    defaultMessage: `Create Expenditure | Colony`,
```
Comment (arrenv) [inline src/utils/hooks/useTitle.ts:146]: I missed this last time, can we please add ` - {colonyName}` here. As Creating an expenditure is in context of a colony.

**Review by arrenv — COMMENTED** [review]
Looks good, thank you for those updates, I have only made one comment of something that I missed previously.

**Review by arrenv — APPROVED** [review]
Looks good, thank you!

File: src/modules/pages/components/ExpenditurePage/ExpenditureTemplate.tsx [hunk 8]
```diff
@@ -0,0 +1,26 @@
+import React, { ReactNode } from 'react';
```
Comment (chinins) [inline src/modules/pages/components/ExpenditurePage/ExpenditureTemplate.tsx:1]: Didn't notice it the first time but I think that this file should be called `ExpenditurePage`. So the file name corresponds to the folder where it is. Remember to update the display name.

**Review by chinins — CHANGES_REQUESTED** [review]
I left one comment about the file name so that we have consistency throughout the dapp and pages folder. Also, please, update the PR name, as requested earlier.

**Review by chinins — APPROVED** [review]
Now all good - nice work!

**Pull request merged**